### PR TITLE
Improve graph visibility and minor cleanups.

### PR DESCRIPTION
**Pull request by porridge**
When adding timeseries to graph, start with higher percentiles, since their
values are usually strictly higher than lower percentiles, which avoids
obscuring graph data.

It also orders data in the onHover labels more naturally.

Swap grey colors for a more visible 90th percentile.

Add .gitignore, missing whitespace and newlines.

### CODE REVIEW

**Comment by porridge** [comment]
Before:

before

After:

after

**Comment by porridge** [comment]
(Note how before the low percentile graphs are obscured by the red 99th percentile graph.)

**Comment by porridge** [comment]
Perhaps @shyamjvs has time to review this?

File: perfdash/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/.idea/
```
Comment (shyamjvs) [inline perfdash/.gitignore]: Guess you can skip the second backslash.

**Comment by shyamjvs** [comment]
Lgtm - Could you squash the commits?

**Comment by porridge** [comment]
Why do you think they should be squashed? They represent separate changes to address independent concerns. I shiver at the mere thought of a commit entitled "Change graph colors and .gitignore file"!

**Comment by shyamjvs** [comment]
As long as they're doing logically different (or significantly disjoint) things, it feels ok to have separate commits. For e.g, a commit for adding whitespaces and another for adding newlines seems unnecessary. If one is looking at the commit history of a file or hunting down for a regression, too many commits feels a bit noisy. You could probably add the gitignore in a separate one, but for the rest the PR title seems appropriate also for the commit msg.

**Comment by porridge** [comment]
With all due respect sir, I actually strongly disagree. :-)

Note that the newlines are added in go files, and the whitespace in the js file, in a different directory, and different layer of the app. Lumping them into a single commit does not seem right.

Also, I have the opposite feeling w.r.t. trawling through `git blame` - it's much easier to recognize and skip a harmless commit when its title says "add newline" than when it's "assorted changes in this general area", at which point you need to scan through it all, and waste time scrolling through pages of nitpicks that the author decided to pack together with their main change.

Another argument for keeping commits small and focused is that they are much easier to roll back. If someone later decides that one of the commits was doing something wrong (say the colors are bad), it can be rolled back without losing the log message formatting changes.

BTW [developer docs](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#2-smaller-is-better-small-commits-small-prs) clearly favor small commits.

Ideally these would be separate PRs, but then again, every PR produces another merge commit, so I tried to strike a balance here.

**Comment by shyamjvs** [comment]
> Note that the newlines are added in go files, and the whitespace in the js file, in a different directory, and different layer of the app. Lumping them into a single commit does not seem right.

When you add whitespaces, fix typos, etc IMO it is ok to change multiple files in the same commit, even if they're at slightly different code paths.

> Also, I have the opposite feeling w.r.t. trawling through git blame - it's much easier to recognize and skip a harmless commit when its title says "add newline" than when it's "assorted changes in this general area", at which point you need to scan through it all, and waste time scrolling through pages of nitpicks that the author decided to pack together with their main change.

Changes to whitespaces, addition of gitignore, etc are relatively low risk changes and I think can be put into one commit (named 'Add gitignore and missing whitespaces in files' or sth) without affecting readability for someone who's skimming through commits. It wouldn't be needed to open the changes if it was a single commit or a bunch of them to know if it's safe to skip. And you probably would never need to rollback such a commit. Besides, having such extra commits can even hurt while mechanical bisection - where you may need some extra rounds.

> Another argument for keeping commits small and focused is that they are much easier to roll back. If someone later decides that one of the commits was doing something wrong (say the colors are bad), it can be rolled back without losing the log message formatting changes.

I agree - but there is a balance on how granular the changes should be. In principle, it might be possible to even change each line in a separate commit (given that it meets some conditions like compilability, correctness, etc). Something like a color change is small enough imo to not need a rollback - it can be changed in another commit.

That said, I generally agree that splitting changes into smaller independent parts is useful for reasons you stated. The line might be slightly blurry :)

**Comment by shyamjvs** [comment]
/lgtm

This might be an interesting discussion for another day.

**Pull request merged**